ui/app/registration/controllers/patientCommonController.js

```diff
@@ -17,7 +17,153 @@ angular.module('bahmni.registration')
             $scope.readOnlyExtraIdentifiers = appService.getAppDescriptor().getConfigValue("readOnlyExtraIdentifiers");
             $scope.showSaveConfirmDialogConfig = appService.getAppDescriptor().getConfigValue("showSaveConfirmDialog");
             $scope.showSaveAndContinueButton = false;
-            $scope.moduleName = appService.getAppDescriptor().getConfigValue('registrationModuleName');
+            $scope.regExtPoints = appService.getAppDescriptor().getExtensions("org.bahmni.registration.identifier", "link");
+
+            $scope.showIframe = false;
+            var buttonMap = new Map();
+            $scope.attributesToBeDisabled = [];
+
+            $scope.openIdentifierPopup = function (identifierType) {
+                var iframe = $document[0].getElementById("identifier-popup");
+                iframe.src = getExtensionPoint(identifierType).src;
+                $scope.showIframe = true;
+                $window.addEventListener("message", function (popupWindowData) {
+                    if (popupWindowData.data.patient !== undefined) {
+                        $rootScope.extenstionPatient = popupWindowData.data.patient;
+                        if ($rootScope.extenstionPatient.id !== undefined) {
+                            if ($rootScope.extenstionPatient.id !== $scope.patient.uuid) {
+                                $window.open(Bahmni.Registration.Constants.existingPatient + $rootScope.extenstionPatient.id, "_self");
+                            }
+                        } else $window.open(Bahmni.Registration.Constants.newPatient, "_self");
+                        $scope.updateInfoFromExtSource($rootScope.extenstionPatient);
+                        $scope.$digest();
+                    }
+                }, false);
+            };
+
+            $scope.isDisabled = function (attribute) {
```

Comment:
maybe better to be more specific function name.. like "isDisabledAttribute()"

Reply:
changed to better function name